src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/MFD/CJ4_MFD.js

```diff
@@ -302,147 +390,1074 @@ class CJ4_FMSContainer extends NavSystemElementContainer {
+
                 this._nextWaypointContainer
-                    .querySelector(".cj4x-navigation-data-waypoint-crs")
-                    .querySelector(".cj4x-navigation-data-waypoint-value")
-                    .textContent = nextWaypoint.bearingInFP.toFixed(1) + " °";
+                    .querySelector(".cj4x-navigation-data-waypoint-eta")
+                    .textContent = nextWaypointETAValue;
+
+                this._destinationWaypointContainer
+                    .querySelector(".cj4x-navigation-data-waypoint-eta")
+                    .textContent = destinationWaypointETAValue;
+
+
+                // Set expected fuel and gross weight
+                if(groundSpeed >= 50){
+                    const fuelFlow = (SimVar.GetSimVarValue("ENG FUEL FLOW PPH:1", "Pounds per hour") + SimVar.GetSimVarValue("ENG FUEL FLOW PPH:2", "Pounds per hour")) / 2;
+                    const expectedFuelUsage = (fuelFlow * (this.calcETEseconds(destinationDistance, groundSpeed) / 3600)).toFixed(0);
+                    const currentFuel = (SimVar.GetSimVarValue("FUEL WEIGHT PER GALLON", "pounds") * SimVar.GetSimVarValue("FUEL TOTAL QUANTITY", "gallons")).toFixed(0);
+                    const expectedFuelAtDestination = (currentFuel - expectedFuelUsage).toFixed(0) < 0 ? 0 : (currentFuel - expectedFuelUsage).toFixed(0);
+                    const grossWeight = SimVar.GetSimVarValue("MAX GROSS WEIGHT", "pounds");
+                    const oilQuantity = SimVar.GetSimVarValue("OIL AMOUNT", "pounds")
+                    const expectedGrossWeight = expectedFuelAtDestination == 0 ? (grossWeight / 1000).toFixed(2) : ((grossWeight - expectedFuelUsage) / 1000).toFixed(2);
+
+                    this._destinationWaypointContainer
+                        .querySelector(".cj4x-navigation-data-waypoint-expected-fuel")
+                        .textContent = expectedFuelAtDestination + " LB " + expectedGrossWeight + " GW";
+
+                }
+
+                if(activeWaypoint){
+                    if(destination.ident == activeWaypoint.ident){
+                        this._destinationWaypointContainer
+                            .setAttribute("style", "color: magenta");
+                        this._activeWaypointContainer
+                            .setAttribute("style", "color: white");
+                    }
+                    else{
+                        this._destinationWaypointContainer
+                            .setAttribute("style", "color: white");
+                        this._activeWaypointContainer
+                            .setAttribute("style", "color: magenta");
+                    }
+                }
             }
-            else {
-                this._nextWaypointContainer.style.display = "none";
+        }
+    }
+    calcETEseconds(distance, currentGroundSpeed) {
+        return (distance / currentGroundSpeed) * 3600;
+    }
+}
+class CJ4_SystemOverlayContainer extends NavSystemElementContainer {
```

Comment:
just wondering if we could pull in the already existing element from asobo here?